pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -178,3 +193,55 @@ func (i *deploymentClientImpl) listDeploymentsForConfig(namespace, configName st
 func (i *deploymentClientImpl) updateDeployment(namespace string, deployment *kapi.ReplicationController) (*kapi.ReplicationController, error) {
 	return i.updateDeploymentFunc(namespace, deployment)
 }
+
+// failureInspection inspects the given deployment configuration for any failure causes
+// and updates the status of the configuration accordingly
+func (c *DeploymentConfigController) failureInspection(config *deployapi.DeploymentConfig, outErr error) error {
```

Comment:
How about making this function just return the message and do the update from the calling side? Makes it easier to test and less side-effecty. I don't like doing the update in multiple places within the function, via defer, etc.